ts/DataGrid/DataGrid.ts

```diff
@@ -234,35 +280,140 @@ class DataGrid {
+    ): void {
+
+        if (overwrite) {
+            this.userOptions = newOptions;
+            this.options = merge(DataGrid.defaultOptions, newOptions);
+        } else {
+            this.userOptions = merge(this.userOptions ?? {}, newOptions);
+            this.options = merge(
+                this.options ?? DataGrid.defaultOptions,
+                this.userOptions
+            );
+        }
+
+        const columnOptionsArray = this.options?.columns;
+        if (!columnOptionsArray) {
+            return;
+        }
+        const columnOptionsObj: Record<string, Column.Options> = {};
+        for (let i = 0, iEnd = columnOptionsArray?.length ?? 0; i < iEnd; ++i) {
+            columnOptionsObj[columnOptionsArray[i].id] = columnOptionsArray[i];
+        }
+
+        this.columnOptionsMap = columnOptionsObj;
+    }
+
+    /**
+     * Updates the data grid with new options. It overwrites the existing column
+     * options array. If you want to merge the column options, use the
+     * `updateColumn` method instead.
      *
      * @param options
-     * The options of the data grid that should be updated.
+     * The options of the data grid that should be updated. If not provided,
+     * the update will be proceeded based on the `this.userOptions` property.
      *
      * @param render
      * Whether to re-render the data grid after updating the options.
+     *
+     * @param overwrite
+     * If true, the column options will be updated by replacing the existing
+     * options with the new ones instead of merging them.
      */
     public async update(
-        options: Options,
-        render: boolean = true
+        options: Options = {},
+        render: boolean = true,
+        overwrite = false
     ): Promise<void> {
-        this.userOptions = merge(this.userOptions, options);
-        this.options = merge(DataGrid.defaultOptions, this.userOptions);
+        this.loadUserOptions(options, overwrite);
 
         let newDataTable = false;
-        if (!this.dataTable || options.dataTable) {
+        if (
+            !overwrite && (!this.dataTable || options.dataTable) ||
+            overwrite && (options.dataTable?.id !== this.dataTable?.id)
+        ) {
             this.loadDataTable(this.options?.dataTable);
             newDataTable = true;
         }
 
         this.querying.loadOptions();
-        await this.querying.proceed(newDataTable);
 
         if (render) {
+            await this.querying.proceed(newDataTable);
             this.renderViewport();
         }
     }
 
+    /**
+     * Updates the column of the data grid with new options. Unlike the
+     * `update` method, it does not overwrite the existing column options array.
+     * Instead, it merges the provided options with the existing ones.
+     *
+     * @param columnId
+     * The ID of the column that should be updated.
+     *
+     * @param options
+     * The options of the columns that should be updated. If null,
```

Comment:
What is the use case for `null` as an option? I find the outcome slightly unexpected.

Reply:
It may happen that the `columns` array contains many options that are not used. `null` simply removes the object from the array with the given ID.

Alternatively, instead, I can make it so that when passing an empty object and the argument `override` is true, the option is removed from the array. 🤔 Let me know what you think.
[`.updateColumn('idToDelete', {}, true, true)` instead of `'idToDelete', null`].

Reply:
@ddragula : you keep saying `override`. You mean `overwrite` ;-)

IMO the last suggestion is better and more intuitive. Quite logical (for me at least) that an empty object with `overwrite:true` will remove it from the array. In the first example I guess the `null` value  would indirectly render `overwrite` useless, and that's "weird". I would go for the last one, even though more verbose.

Aaaaaand, somewhere in a bright and sunny future we will see `.deleteColumnOptions(idToDelete)` 😍 😂😂

Reply:
>you keep saying override. You mean overwrite ;-)

I can't unlearn it 🙈 

I'll implement this then, because I actually like the idea much more now. Why didn't I come up with that sooner? 🤔